pyanaconda/modules/payloads/payload/rpm_ostree/installation.py

```diff
@@ -0,0 +1,342 @@
+                bindopt = '--rbind'
+            else:
+                bindopt = '--bind'
+            safe_exec_with_redirect("mount", [bindopt, src, dest])
+        self._internal_mounts.append(src if bind_ro else dest)
+
+    def run(self):
+        """Run the task.
+
+        :return: list of bindmounts created for internal use
+        :rtype: list of str
+        """
+        device_tree = STORAGE.get_proxy(DEVICE_TREE)
+        mount_points = device_tree.GetMountPoints()
+
+        # Currently, blivet sets up mounts in the physical root.
+        # We used to unmount them and remount them in the sysroot, but
+        # since 664ef7b43f9102aa9332d0db5b7d13f8ece436f0 we now just set up
+        # bind mounts.
+
+        # Make /usr readonly like ostree does at runtime normally
+        self._setup_internal_bindmount('/usr', bind_ro=True, src_physical=False)
+
+        # Explicitly do API mounts; some of these may be tracked by blivet, but
+        # we'll skip them below.
+        api_mounts = ["/dev", "/proc", "/run", "/sys"]
+        for path in api_mounts:
+            self._setup_internal_bindmount(path)
+
+        # Handle /var; if the admin didn't specify a mount for /var, we need
+        # to do the default ostree one.
+        # https://github.com/ostreedev/ostree/issues/855
+        var_root = '/ostree/deploy/' + self._source_config.osname + '/var'
+        if mount_points.get("/var") is None:
+            self._setup_internal_bindmount(var_root, dest='/var', recurse=False)
+        else:
+            # Otherwise, bind it
+            self._setup_internal_bindmount('/var', recurse=False)
+
+        # Now that we have /var, start filling in any directories that may be
+        # required later there. We explicitly make /var/lib, since
+        # systemd-tmpfiles doesn't have a --prefix-only=/var/lib. We rely on
+        # 80-setfilecons.ks to set the label correctly.
+        mkdirChain(conf.target.system_root + '/var/lib')
+        # Next, run tmpfiles to make subdirectories of /var. We need this for
+        # both mounts like /home (really /var/home) and %post scripts might
+        # want to write to e.g. `/srv`, `/root`, `/usr/local`, etc. The
+        # /var/lib/rpm symlink is also critical for having e.g. `rpm -qa` work
+        # in %post. We don't iterate *all* tmpfiles because we don't have the
+        # matching NSS configuration inside Anaconda, and we can't "chroot" to
+        # get it because that would require mounting the API filesystems in the
+        # target.
+        for varsubdir in ('home', 'roothome', 'lib/rpm', 'opt', 'srv',
+                          'usrlocal', 'mnt', 'media', 'spool', 'spool/mail'):
+            safe_exec_with_redirect("systemd-tmpfiles",
+                                    ["--create", "--boot", "--root=" + conf.target.system_root,
+                                     "--prefix=/var/" + varsubdir])
+
+        # Handle mounts like /boot (except avoid /boot/efi; we just need the
```

Comment:
I would consider to split the `run` method into smaller chunks. The `run` method could just call a series of private methods. It would be nice to move the `run` method on the top in that case.